stores/listingCart.ts

```diff
@@ -0,0 +1,91 @@
+import { defineStore } from 'pinia'
+import { EntityWithId, NFTMetadata } from '@/components/rmrk/service/scheme'
+import { ComputedRef } from 'vue'
+import type { Prefix } from '@kodadot1/static'
+
+export type ListCartItem = {
+  id: string
+  name: string
+  urlPrefix: string
+  price: string
+  listPrice: number | null
+  collection: EntityWithId
+  meta?: NFTMetadata
+  metadata?: string
+}
+
+type ID = string
+
+interface State {
+  items: ListCartItem[]
+  unlistedOwned: ListCartItem[]
+  chain: ComputedRef<Prefix>
+  decimals: ComputedRef<number>
+}
+
+const localStorage = useLocalStorage<ListCartItem[]>('listingCart', [])
+export const useListingCartStore = defineStore('listingCart', {
+  state: (): State => ({
+    items: localStorage.value,
+    unlistedOwned: [],
+    chain: usePrefix().urlPrefix,
+    decimals: useChain().decimals,
+  }),
+  getters: {
+    itemsInChain: (state): ListCartItem[] =>
+      state.items.filter((item) => item.urlPrefix === state.chain),
+    count() {
+      return this.itemsInChain.length
+    },
+    incompleteListPrices: (state) =>
+      state.items.filter((item) => !item.listPrice).length,
+  },
+  actions: {
+    getItem(id: ID) {
+      return this.items.find((item) => item.id === id)
+    },
+    isItemInCart(id: ID) {
+      return this.existInItemIndex(id) !== -1
+    },
+    existInItemIndex(id: string, items?: ListCartItem[]) {
+      return (items ?? this.items).findIndex((item) => item.id === id)
+    },
```

Comment:
I don't think this belongs in the store

the store should have either
1. getters : read store value
2. setter : modify store value

this is neither

Reply:
Where can we place this?